server/src/core/agent/cache.py

```diff
@@ -1,21 +1,26 @@
-from core.common.encode import encode_to_base64, decode_from_base64
-from core.common.cache import CacheMemory
+# from core.common.encode import encode_to_base64, decode_from_base64
+from core.common import conn
 
 
+# TODO: Why were there used base64 encodings?
+#       Experiments show that it receives binary strings as arguments and works fine.
 class PredictCache:
     def __init__(self):
-        self.cache = CacheMemory(30)
+        # self.cache = CacheMemory(30)  # TODO: Limit the number of cached entries?
+        self.redis = conn.get_redis_instance()  # TODO: Rewrite with dependency injection.
 
-    def put(self, key, value: str):
-        k = encode_to_base64(key)
-        v = encode_to_base64(value)
-        self.cache.put(k, v)
+    async def put(self, key, value: str):
+        # k = encode_to_base64(key)
+        # v = encode_to_base64(value)
+        await self.redis.hset("predict_cache:" + key, "value", value)
```

Comment:
Is this correct?  I think it should be value = value

```
k = encode_to_base64(key)
await self.redis.hset("predict_cache:" + k, value=value)
```

Reply:
`redis_server.hset(key, value, number)` in [this StackOverflow answer](https://stackoverflow.com/a/24017598) suggests that it is correct. Moreover, I tested that it really sets the key by using tracing and it worked. So, `redis.hset("predict_cache:" + key, "value", value)` is correct.

(I suspect that `value=value` may work, too, but checking this would be a superfluous experimentation.)